analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs

```diff
@@ -28,16 +28,15 @@ namespace SonarAnalyzer.Helpers;
 
 public class SonarLintXmlReader
 {
-    public static readonly SonarLintXmlReader Empty = new(null, LanguageNames.CSharp);
+    public static readonly SonarLintXmlReader Empty = new(null);
 
     private readonly SonarLintXml sonarLintXml;
-    private readonly string propertyLanguage;
 
     private bool? ignoreHeaderComments;
-    public bool IgnoreHeaderComments => ignoreHeaderComments ??= ReadBoolean(ReadSettingsProperty($"sonar.{propertyLanguage}.ignoreHeaderComments"));
+    public bool IgnoreHeaderComments(string language = LanguageNames.CSharp) => ignoreHeaderComments ??= ReadBoolean(ReadSettingsProperty($"sonar.{LanguageXmlName(language)}.ignoreHeaderComments"));
```

Comment:
The default parameter doesn't make sense to me. Production code doesn't need it. 
```suggestion
    public bool IgnoreHeaderComments(string language) => ignoreHeaderComments ??= ReadBoolean(ReadSettingsProperty($"sonar.{LanguageXmlName(language)}.ignoreHeaderComments"));
```